test/e2e/common/expansion.go

```diff
@@ -709,7 +710,7 @@ func waitForPodContainerRestart(f *framework.Framework, pod *v1.Pod, probeFilePa
 
 	// Fix liveness probe
 	By("Rewriting the file")
-	out, err = framework.RunKubectl("exec", fmt.Sprintf("--namespace=%s", pod.Namespace), pod.Name, "--container", pod.Spec.Containers[1].Name, "--", "/bin/sh", "-c", fmt.Sprintf("echo test-after > %v", probeFilePath))
+	out, err = framework.RunKubectl("exec", fmt.Sprintf("--namespace=%s", pod.Namespace), pod.Name, "--container", pod.Spec.Containers[1].Name, "--", "/bin/sh", "-c", fmt.Sprintf("echo test-after > %v", subPathMount))
```

Comment:
Apologies. Fixed it so that it tests for the remounted volume and not just the subpath

Reply:
@msau42 The pod not coming ready error seems to have resurfaced after a period of success. I cannot see anything obvious in the alpha suite which would cause the pod to not become ready. Any clues? Updating the annotation should just be enough?

Reply:
I've updated this state failure test to use the container restart mechanism after a failed condition - this then keeps the restarts consistent.